_posts/2016-10-08-jumping-into-elasticsearch.markdown

```diff
@@ -0,0 +1,348 @@
+
+### Creating an index
+First, let's create an index for a bare-bones social network. In the lefthand Sense panel paste (or type, and take advantage of the awesome IntelliSense):
+
+```
+PUT /socialnetwork
+{
+  "mappings": {
+    "user": {
+      "properties": {
+        "first_name": {
+          "type": "string"
+        },
+        "last_name": {
+          "type": "string"
+        },
+        "city": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "gender": {
+          "type": "string"
+        },
+        "age": {
+          "type": "integer"
+        },
+        "skills": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+```
+
+As you might as have guessed from the above JSON blob, in addition to creating the `socialnetwork` index, we explicitly mapped a `user` type. But I'll let you in on a secret: we could have left out the request object (`PUT /socialnetwork`), then simply created a `user` document and let Elasticsearch handle the mapping *dynamically*. We could have even started creating documents without initially creating the index! All we'd have to do is change the path to `/socialnetwork/user`, and provide the user data in the body straightaway.
+
+So with that said, let's ditch this thing by running `DELETE /socialnetwork`, and start from scratch.
+
+### Bulk API
+Now we'll take advantage of the `_bulk` API to create two new indexes and upload a ton of documents all at once.
+
+First, we need to <a href='/assets/data/mock-data.json' download>download</a> some mock data in raw JSON format. (<a href='https://raw.githubusercontent.com/TechnologyAdvice/technologyadvice.github.io/master/assets/data/mock-data.json' target='_blank'>preview</a>)
+
+Then in the command line, navigate to where the data is stored, and upload it to Elasticsearch:
+
+```
+curl -XPOST <your docker host ip>:9200/_bulk --data-binary "@mock-data.json"
+```
+
+(You'll see a huge blob of JSON full of the newly created documents.)
+
+We left out the index and type before `/_bulk` in the above URI, because the file included documents for separate indexes, which were distinguished like this:
+
+```
+{"index":{"_index":"socialnetwork","_type":"user","_id":"1"}}
+... user document data (on one line because of the `--data-binary` option)...
+{"index":{"_index":"localsearch","_type":"business","_id":"1"}}
+... business document data ...
+```
+
+You can generally narrow the scope of your requests by including the index and/or type in the URI.
+
+Each document also has a simple integer id, but if no id is specified on creation, a unique alphanumeric string will be generated automatically.
+
+You can perform basic CRUD on any document by specifying its id in the URI, such as `GET /socialnetwork/user/1`.
```

Comment:
This is a lot of raw ES facts in a row -- it would be good for pacing to prompt the reader to actually hit that endpoint, and show what comes back in the result.